### ISSUE TITLE
Contribute with translation

### ISSUE THREAD

**Issue by andrekutianski** [castlamp/zenbership#24]
Hello! It's possible contribute with translation of Zenbership? My interesting is translate to Brazilian portuguese (search on web but don't find any translation).

**Comment by jbelelieu** [comment]
Translations are always welcome, however at the present time I would ask that you wait for v109 stable to be released for a number of reasons:

1. There are plans to change the default theme/templates to a bootstrap-like framework. Since there are no official "language packs" with Zenbership, all translations would have to be done directly on the templates themselves. If we end up changing the templates, your work would be lost with the release of v109 stable.

2. Likewise, the admin control panel doesn't have "language packs" either, which means all translations would have to be hard coded. This isn't ideal and I would want a better language tool in place before committing to having users to translation work.

**Comment by itworksjbo** [comment]
Hi! I'm eager to contribute a German translation as well. Do you by any chance have a rough ETA on the stable release yet?

**Comment by jbelelieu** [comment]
First of all, thank you! It's truly appreciated!

The good new is we're getting closer and closer to a stable release. There are just a couple bugs that need to be resolved before we can call this release "stable". However overall work is progressing very quickly now, and modernization of the templates has begun to make them "adapt" to existing web sites better. Once that step is completed, we can start working on translations! I'll keep you updated. Follow the blog for the latest news on releases.

**Comment by itworksjbo** [comment]
I found a bug in the alpha version regarding the language selection. I copied /pp-templates/html/zoid/en to /pp-templates/html/zoid/de and changed the default language from english to german in the admin options. This results in a blank website unless adding the ?lang=de.

**Comment by jbelelieu** [comment]
This will be resolved shortly. The issue is a type-o in the database which uses a two letter code other than "de" for german. It is set to "dr" currently which is obviously wrong. Update to follow in the next commit for this! Sorry!!

**Comment by itworksjbo** [comment]
No worries, thanks for the fix. You're doing great work here and I'm happy if I can help. The one or two bugs here are nothing compared to the cr*p I had to endure with CRM/Member Management/Event Management Software I was forced to use up to date. It's so buggy, it's barely usable and paid "Support" takes months to get anything done, even when I told them what exactly was causing the problems. And closed source. Can't wait to get zenbership up and running on the first productive systems once the stable is released and I get to translate it. 👍

**Comment by MatBje** [comment]
Hi Jon or anyone on the dev team,

I'd like to try my hand at translating Zenbership member pages into Swedish.

So far I have found three places that seem to require changes for the user experience to change fully into Swedish:

1. Add the value "sv:Swedish" in the table "ppSD_options" on the rwo with "id = language" in the field named "options". 
UPDATE `ppSD_options` SET `options` = 'ar:Arabic|bn:Bengali|zh:Chinese|de:Deutsch|en:English|es:Espanol|fr:Francais|hi:Hindi|ja:Japanese|pt:Portuguese|pa:Punjabi|ru:Russian|sv:Swedish' WHERE `ppSD_options`.`id` = 'language';

This can be added on installation by editing the file "setup\mysql\inserts.php" and adding the value there on line 729.

2. Copy /pp-templates/html/zoid/en to /pp-templates/html/zoid/sv and translate the files.

3. Add all error messages in Swedish to the table "ppSD_error_codes".

These could be added to the setup also by editing the "INSERT INTO `ppSD_error_codes" section.
Make sure to add the language code as a new column and specify this for all the original English messages too.


Are there any more places where changes need to be made to affect the member experience i.e. not the control panel, just anything a normal user might see or receive as an e-mail or sms?

I should also add that to copy the files from a Windows computer using FileZilla to a Linux server, Binary mode should be selected for all file transfers, not Auto as this will cause Error 500 when setting up the system. This has been noted previously by others for earlier versions but the installation instructions for V115 could do with an update to reflect this too.

Thanks! It is a fantastically feature rich and well structured system with everything we could dream of so it would be great to have a Swedish version too.
Best, /Mats

**Comment by FolkertRA** [comment]
I am interested in using this in Dutch and am happy to do some translations for that as soon as the new version is available.

**Comment by smellems** [comment]
I'm also interested in contributing French translations to the default themes.  For the HTML, I copied `en` to `/pp-templates/html/zoid/fr`  I noticed that some files use variables like %page_title%, where does that come from?  I can't seem to find the pages..

I also have same question as @MatBje about translating emails templates.

Thanks!

**Comment by jbelelieu** [comment]
That would be the Page Title for the content piece the program is rendering. "Content > Pages"

**Comment by jbelelieu** [comment]
For emails you would have to create a new theme altogether and use that theme. The theme takes on the language into which you are translating it.

**Comment by smellems** [comment]
> That would be the Page Title for the content piece the program is rendering. "Content > Pages"

Makes sense. This is a fresh install and there are no page created.  Where is page_title for News Homepage, Viewing Calendar, Shop, Cart..?

**Comment by jbelelieu** [comment]
In those cases the template contains the meta data.

"INTEGRATION > HTML TEMPLATES"
Edit any template to see the meta data.

**Comment by ITfee** [comment]
Does Zenbership support language packs since this topic started? I want to contribute to a german translation but don't want to hard code it just to loose it after the next update.
